community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/planner/logical/plans/LogicalPlan.scala

```diff
@@ -68,6 +68,16 @@ abstract class LogicalPlan
     }
   }
 
+  def copyPlan(): LogicalPlan = {
+    try {
+      val arguments = this.children.toList :+ solved
+      copyConstructor.invoke(this, arguments: _*).asInstanceOf[this.type]
```

Comment:
I really don't like this use of a reflective copyConstructor but it has been around a while so let's try to get rid of it in a separate PR